src/core/app-template.cc

```diff
@@ -109,12 +100,33 @@ app_template::app_template(app_template::config cfg)
 {
 }
 
+void
+app_template::add_extra_options(boost::program_options::options_description& opts) {
```

Comment:
instead of mutating the parameter, i'd suggest returning a `boost::program_options::options_description`. more readable this way, IMHO. BTW, could make this function `static`, if we don't want to mark it with the `const` specifier.